packages/react-dom/src/__tests__/ReactDOMFragmentRefs-test.js

```diff
@@ -0,0 +1,557 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails reactcore
+ */
+
+'use strict';
+
+let React;
+let ReactDOMClient;
+let act;
+let container;
+let Fragment;
+let Activity;
+
+describe('FragmentRefs', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    Fragment = React.Fragment;
+    Activity = React.unstable_Activity;
+    ReactDOMClient = require('react-dom/client');
+    act = require('internal-test-utils').act;
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  // @gate enableFragmentRefs
+  it('attaches a ref to Fragment', async () => {
+    const fragmentRef = React.createRef();
+    const root = ReactDOMClient.createRoot(container);
+
+    await act(() =>
+      root.render(
+        <div id="parent">
+          <Fragment ref={fragmentRef}>
+            <div id="child">Hi</div>
+          </Fragment>
+        </div>,
+      ),
+    );
+    expect(container.innerHTML).toEqual(
+      '<div id="parent"><div id="child">Hi</div></div>',
+    );
+
+    expect(fragmentRef.current).not.toBe(null);
+  });
+
+  // @gate enableFragmentRefs
+  it('accepts a ref callback', async () => {
+    let fragmentRef;
+    const root = ReactDOMClient.createRoot(container);
+
+    await act(() => {
+      root.render(
+        <Fragment ref={ref => (fragmentRef = ref)}>
+          <div id="child">Hi</div>
+        </Fragment>,
+      );
+    });
+
+    expect(fragmentRef.current).not.toEqual(null);
+  });
+
+  // @gate enableFragmentRefs
+  it('is available in effects', async () => {
+    function Test() {
+      const fragmentRef = React.useRef(null);
+      React.useLayoutEffect(() => {
+        expect(fragmentRef.current).not.toBe(null);
+      });
+      React.useEffect(() => {
+        expect(fragmentRef.current).not.toBe(null);
+      });
+      return (
+        <Fragment ref={fragmentRef}>
+          <div />
+        </Fragment>
+      );
+    }
+
+    const root = ReactDOMClient.createRoot(container);
+    await act(() => root.render(<Test />));
+  });
+
+  describe('focus()', () => {
+    // @gate enableFragmentRefs
+    it('focuses the first focusable child', async () => {
```

Comment:
Another interesting case would be how it interacts with hoistables (e.g. `<style />`) and `createPortal`.

Reply:
Added a style tag before the focusable element